e2e/uimodel/site/siteservers.go

```diff
@@ -0,0 +1,307 @@
+	Expect(p.site.page.FindByClass("grv-site-servers-btn-start").Click()).
+		To(Succeed(), "should start expand operation")
+
+	p.site.WaitForOperationCompletion()
+	utils.PauseForServerListRefresh()
+	updatedItems := p.GetSiteServers()
+
+	var newItem SiteServer
+	for i, item := range updatedItems {
+		for _, existingItem := range currentItems {
+			if item.AdvertiseIP == existingItem.AdvertiseIP {
+				break
+			}
+			newItem = updatedItems[i]
+		}
+	}
+
+	Expect(newItem).ToNot(BeNil(), "should find a new server in the server list")
+	return newItem
+}
+
+func (p *ServerPage) hasServer(server SiteServer) func() bool {
+	return func() bool {
+		for _, existingServer := range p.GetSiteServers() {
+			if existingServer.Hostname == server.Hostname {
+				return true
+			}
+		}
+		return false
+	}
+}
+
+func (p *ServerPage) clickDeleteServer(serverId string) {
+	const scriptTemplate = `
+            var targetIndex = -1;
+            var rows = document.querySelectorAll(".grv-site-servers .grv-table .dropdown-toggle");
+            rows.forEach( (z, index) => {
+                if( z.innerText.indexOf("%v") !== -1) targetIndex = index; 
+            })
+            
+            return targetIndex;
+        `
+	var result int
+	page := p.site.page
+
+	script := fmt.Sprintf(scriptTemplate, serverId)
+	page.RunScript(script, nil, &result)
+
+	result = result + 1
+	actionsMenuPath := fmt.Sprintf(".grv-site-servers tr:nth-child(%v) .dropdown-toggle", result)
+	Expect(page.Find(actionsMenuPath).Click()).To(
+		Succeed(),
+		"should find and expand action menu")
+
+	deleteActionBtnPath := fmt.Sprintf(".grv-site-servers tr:nth-child(%v) .dropdown-menu .fa-trash", result)
+	Expect(page.Find(deleteActionBtnPath).Click()).To(
+		Succeed(),
+		"should find and click on server delete action")
+}
+
+func (p *ServerPage) getFirstAvailableAWSInstanceType() string {
+	var instanceType string
+	const js = `
+		var cssSelector = ".grv-site-servers-provisioner-new-instance-type li a"; 
+		var items = document.querySelectorAll(cssSelector)			
+		return items.length > 0 ? items[0].text : "";  			
+	`
+
+	Expect(p.site.page.RunScript(js, nil, &instanceType)).To(
+		Succeed(),
+		"should retrieve first available instance type")
+
+	Expect(instanceType).NotTo(BeEmpty(), "cannot find any instance types")
+
+	return instanceType
+}
+
+func (p *ServerPage) selectOnPremProfile(profileName string) {
+	log.Infof("selecting onprem profile")
+	if profileName == "" {
+		Expect(p.site.page.FindByClass("grv-control-radio-indicator").Click()).To(
+			Succeed(),
+			"should select first available profile")
+		return
+	}
+
+	profileLabel := p.getProfileLabel(profileName)
+	utils.SelectRadio(p.site.page, ".grv-control-radio", func(value string) bool {
+		return strings.HasPrefix(value, profileLabel)
+	})
+}
+
+func (p *ServerPage) getProfileLabel(profileName string) string {
+	var profileLabel string
+	siteName := p.site.domainName
+	const jsTemplate = `
+		var profileName = "%v";
```

Comment:
I'd actually like this to be a text/template - this is hard to see where which variables go.